### DISCUSSION TITLE
custom node shape/symbol in diagram chart?

### DISCUSSION THREAD

**Discussion by retsyo** [mermaid-js/mermaid#3660] [Q&A]
is it possible to let user choose symbols for node, for example, cloud as the attachment(from https://www.vectorstock.com/royalty-free-vector/cloud-flowchart-charts-vector-1013538) shows?

Since mermaid has introduced such thing, i.e. round corner rectangle by `id1(This is the text in the box)`, maybe more can be used, for example, since `:?:` is not used often in normal text, we can write `I am a cloud :?: custom.cloud.cloud1` to use `cloud1` in `custom.cloud` hiberarchy defined in JSON or local direcory/file system.

Background, currently I try to express my diagram which uses cloud service, so a cloud-like symbol is more suitable than `id1[(Database)]` (a cylindrical shape)

If mermaid does not/will not support such feature, could you please name some tools? I mean a tool in which user can write pure text then render the final diagram, not a traditional drawing tool which needs drag-and-drop.

thanks

![cloud_flowchart](https://user-images.githubusercontent.com/7960913/195470459-27dcc8dc-56de-4d01-ae9c-aedee687c146.jpg)

**Comment by weedySeaDragon** [comment]
The existing shapes used are all simple SVG shapes. If there is a simple, _completely **copyright** free_ cloud SVG diagram, then it could be added to the code if Knut S. agrees to it.

I don't know if there are any plans in the future to support users adding their own graphic symbols.

**Comment by philCryoport** [comment]
@weedySeaDragon I'd like to have an email node shape to indicate someone or something will send an email message.

Example uses:
<img width="121" alt="image" src="https://github.com/user-attachments/assets/f700237c-d3d3-4c59-b7b3-c6c60a6e08dc">

<img width="97" alt="image" src="https://github.com/user-attachments/assets/fb226bb2-48a7-4ce4-839a-ca48e5a0798d">

[PlantUML offers the ability to use custom icons -- with a large selection of libraries](https://plantuml.com/stdlib).  

[Here's the Office 2014 email icon](https://github.com/Roemer/plantuml-office/blob/master/office2014/Concepts/email.png):

![image](https://github.com/user-attachments/assets/622c11bb-d8c2-4702-b27d-cc143eab5b9b)

I might be blind, but I don't see MermaidJS offering that feature.  Did I miss it somewhere?

**Reply by weedySeaDragon** [reply]
It's really up to @knsv .

**Comment by philCryoport** [comment]
Looks like one can insert `<img>` tags within blocks:

```
flowchart TD
    A[Christmas<img src='http://www.pngall.com/wp-content/uploads/2017/05/Star-PNG-Image-File.png'/> ] -->|Get money| B(Go shopping)
    B --> C{Let me think}
    C -->|One| D[Laptop]
    C -->|Two| E[iPhone]
    C -->|Three| F[fa:fa-car Car]
```
